pkgs/tools/package-management/fortran-fpm/default.nix

```diff
@@ -0,0 +1,46 @@
+{ stdenv, lib, pkgs, fetchurl }:
+
+let
+  pname = "fortran-fpm";
+
+  version = "0.9.0";
+
+  src = fetchurl {
+    url = "https://github.com/fortran-lang/fpm/releases/download/v${version}/fpm-${version}.F90";
+    sha256 = "sha256-VWs4g7odtv1iyZunFD8el+u0CXKcQgnwOqPG/JcMzj8=";
+  };
+
+  buildPath = "build/bootstrap";
+in
```

Comment:
Thanks for the info; do you have a link to discussion about this? I'd like to make sure we update the relevant parts of the nixpkgs manual with the style guide

Reply:
IIRC, the motivation for using let bindings was because recursive attribute sets in `mkDerviation` had weird semantics with overrides. Not sure how let bindings made this better though.

Anyways, nowadays we can call `mkDerivation` with a recursive function instead, which does have the proper behavior with overrides.